metadata_backend/api/handlers/submission.py

```diff
@@ -313,3 +314,61 @@ async def put_submission_path(self, req: Request) -> Response:
         body = ujson.dumps({"submissionId": upd_submission}, escape_forward_slashes=False)
         LOG.info("PUT %r in submission with ID: %r was successful.", schema, submission_id)
         return web.Response(body=body, status=200, content_type="application/json")
+
+    async def get_submission_files(self, req: Request) -> Response:
+        """Get files from a submission with the version present in submission.
+
+        :param req: GET request
+        :returns: HTTP No Content response
+        """
+        submission_id = req.match_info["submissionId"]
+        db_client = req.app["db_client"]
+        submission_operator = SubmissionOperator(db_client)
+
+        # Check submission exists and is not already published
+        await submission_operator.check_submission_exists(submission_id)
+        await submission_operator.check_submission_published(submission_id, req.method)
+
+        await self._handle_check_ownership(req, "submission", submission_id)
+
+        file_operator = FileOperator(db_client)
+
+        files = await file_operator.read_submission_files(submission_id)
+
+        LOG.info("GET files for submission with ID: %r was successful.", submission_id)
+        return web.Response(
+            body=ujson.dumps(files, escape_forward_slashes=False), status=200, content_type="application/json"
+        )
+
+    async def add_submission_files(self, req: Request) -> Response:
+        """Add files to a submission.
+
+        Body needs to contain a list of files with accessionId and version.
+
+        :param req: POST request with metadata schema in the body
+        :returns: HTTP No Content response
+        """
+        submission_id = req.match_info["submissionId"]
+        db_client = req.app["db_client"]
+        submission_operator = SubmissionOperator(db_client)
+
+        # Check submission exists and is not already published
+        await submission_operator.check_submission_exists(submission_id)
+        await submission_operator.check_submission_published(submission_id, req.method)
+
+        await self._handle_check_ownership(req, "submission", submission_id)
+
+        file_operator = FileOperator(db_client)
+
+        data: List[Dict] = await req.json()
+
+        if all("accessionId" and "version" in d for d in data):
```

Comment:
I think this needs to be: `"accessionId" in d and "version" in d for d in data`

Reply:
good catch, will fix.